packages/btcindexer/src/index.ts

```diff
@@ -35,6 +37,31 @@ export default {
 		}
 	},
 
+	async queue(
+		batch: MessageBatch<BlockQueueMessage>,
+		env: Env,
+		_ctx: ExecutionContext,
+	): Promise<void> {
+		console.log(`Processing batch of ${batch.messages.length} messages from ${batch.queue}`);
+		const nbtcAddresses = await fetchNbtcAddresses(env.DB);
+		const nbtcAddressesMap = new Map<string, NbtcAddress>(
+			nbtcAddresses.map((addr) => [addr.btc_address, addr]),
+		);
+		const storage = new CFStorage(env.DB, env.btc_blocks, env.nbtc_txs);
+		const indexer = await indexerFromEnv(env, nbtcAddressesMap);
+
+		for (const message of batch.messages) {
+			try {
+				await storage.insertBlockFromQueue(message.body);
+				await indexer.processBlock(message.body);
+				await message.ack();
+			} catch (e) {
+				console.error("Failed to process block", toSerializableError(e));
+				await message.retry();
+			}
+		}
+	},
```

Comment:
extract this to a helper function in a new file. The function should take only necessary arguments: storage and indexer

Reply:
done